Persistent caching for Models fails to enable for MySQL

**Describe the bug**
Persistent caching for Models fails to enable for MySQL. It works for MariaDB.

**To Reproduce**
1. Admin > Settings > Caching > enable Model caching
2. Admin > Databases > create connection to MySQL 5.7 or 8.0
3. After sync, go to the database and click "Turn model caching on" - fails with `Lack permission to read table in schema metabase_cache_f74b7_2`
![image](https://user-images.githubusercontent.com/1447303/191558003-59649a2a-704a-4175-83df-f140f4004707.png)

<details><summary>Full stacktrace</summary>

```
2022-09-21 18:22:36,035 INFO mysql.ddl :: Step :persist.check/create-schema was successful for db DELETE
2022-09-21 18:22:36,038 WARN mysql.ddl :: #error {
 :cause You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'values (1)' at line 2
 :via
 [{:type java.sql.SQLSyntaxErrorException
   :message (conn=4) You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'values (1)' at line 2
   :at [org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory createException ExceptionFactory.java 62]}
  {:type org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException
   :message You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'values (1)' at line 2
   :at [org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException of MariaDbSqlException.java 34]}
  {:type java.sql.SQLException
   :message You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'values (1)' at line 2
   :at [org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol readErrorPacket AbstractQueryProtocol.java 1695]}]
 :trace
 [[org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol readErrorPacket AbstractQueryProtocol.java 1695]
  [org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol readPacket AbstractQueryProtocol.java 1557]
  [org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol getResult AbstractQueryProtocol.java 1520]
  [org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol executeQuery AbstractQueryProtocol.java 318]
  [org.mariadb.jdbc.ClientSidePreparedStatement executeInternal ClientSidePreparedStatement.java 220]
  [org.mariadb.jdbc.ClientSidePreparedStatement execute ClientSidePreparedStatement.java 149]
  [org.mariadb.jdbc.ClientSidePreparedStatement executeQuery ClientSidePreparedStatement.java 163]
  [com.mchange.v2.c3p0.impl.NewProxyPreparedStatement executeQuery NewProxyPreparedStatement.java 1471]
  [clojure.java.jdbc$execute_query_with_params invokeStatic jdbc.clj 1090]
  [clojure.java.jdbc$execute_query_with_params invoke jdbc.clj 1084]
  [clojure.java.jdbc$db_query_with_resultset_STAR_ invokeStatic jdbc.clj 1106]
  [clojure.java.jdbc$db_query_with_resultset_STAR_ invoke jdbc.clj 1093]
  [clojure.java.jdbc$query invokeStatic jdbc.clj 1182]
  [clojure.java.jdbc$query invoke jdbc.clj 1144]
  [clojure.java.jdbc$query invokeStatic jdbc.clj 1160]
  [clojure.java.jdbc$query invoke jdbc.clj 1144]
  [metabase.driver.sql.ddl$jdbc_query invokeStatic ddl.clj 23]
  [metabase.driver.sql.ddl$jdbc_query invoke ddl.clj 20]
  [metabase.driver.mysql.ddl$exec_async$fn__82846$fn__82850 invoke ddl.clj 21]
  [metabase.driver.mysql.ddl$exec_async$fn__82846 invoke ddl.clj 17]
  [clojure.core.async$thread_call$fn__16312 invoke async.clj 484]
  [clojure.lang.AFn run AFn.java 22]
  [java.util.concurrent.ThreadPoolExecutor runWorker ThreadPoolExecutor.java 1128]
  [java.util.concurrent.ThreadPoolExecutor$Worker run ThreadPoolExecutor.java 628]
  [java.lang.Thread run Thread.java 829]]}
2022-09-21 18:22:36,038 INFO mysql.ddl :: Step :persist.check/create-table was successful for db DELETE
Exception in thread "async-thread-macro-13" java.sql.SQLSyntaxErrorException: (conn=4) You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'values (1)' at line 2
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:62)
	at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158)
	at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:262)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:229)
	at org.mariadb.jdbc.ClientSidePreparedStatement.execute(ClientSidePreparedStatement.java:149)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeQuery(ClientSidePreparedStatement.java:163)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:1471)
	at clojure.java.jdbc$execute_query_with_params.invokeStatic(jdbc.clj:1090)
	at clojure.java.jdbc$execute_query_with_params.invoke(jdbc.clj:1084)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1106)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.sql.ddl$jdbc_query.invokeStatic(ddl.clj:23)
	at metabase.driver.sql.ddl$jdbc_query.invoke(ddl.clj:20)
	at metabase.driver.mysql.ddl$exec_async$fn__82846$fn__82850.invoke(ddl.clj:21)
	at metabase.driver.mysql.ddl$exec_async$fn__82846.invoke(ddl.clj:17)
	at clojure.core.async$thread_call$fn__16312.invoke(async.clj:484)
	at clojure.lang.AFn.run(AFn.java:22)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'values (1)' at line 2
	at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:194)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:177)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:321)
	at org.mariadb.jdbc.ClientSidePreparedStatement.executeInternal(ClientSidePreparedStatement.java:220)
	... 20 more
Caused by: java.sql.SQLException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'values (1)' at line 2
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1695)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1557)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1520)
	at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:318)
	... 21 more
2022-09-21 18:22:36,040 WARN mysql.ddl :: Error in `:persist.check/read-table` while checking for model persistence permissions.
2022-09-21 18:22:36,042 WARN mysql.ddl :: #error {
 :cause Table 'metabase_cache_f74b7_93.persistence_check_2786' doesn't exist
 :via
 [{:type java.sql.SQLSyntaxErrorException
   :message (conn=5) Table 'metabase_cache_f74b7_93.persistence_check_2786' doesn't exist
   :at [org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory createException ExceptionFactory.java 62]}
  {:type org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException
   :message Table 'metabase_cache_f74b7_93.persistence_check_2786' doesn't exist
   :at [org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException of MariaDbSqlException.java 34]}
  {:type java.sql.SQLException
   :message Table 'metabase_cache_f74b7_93.persistence_check_2786' doesn't exist
   :at [org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol readErrorPacket AbstractQueryProtocol.java 1695]}]
 :trace
 [[org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol readErrorPacket AbstractQueryProtocol.java 1695]
  [org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol readPacket AbstractQueryProtocol.java 1557]
  [org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol getResult AbstractQueryProtocol.java 1520]
  [org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol executeQuery AbstractQueryProtocol.java 318]
  [org.mariadb.jdbc.ClientSidePreparedStatement executeInternal ClientSidePreparedStatement.java 220]
  [org.mariadb.jdbc.ClientSidePreparedStatement execute ClientSidePreparedStatement.java 149]
  [org.mariadb.jdbc.ClientSidePreparedStatement executeQuery ClientSidePreparedStatement.java 163]
  [com.mchange.v2.c3p0.impl.NewProxyPreparedStatement executeQuery NewProxyPreparedStatement.java 1471]
  [clojure.java.jdbc$execute_query_with_params invokeStatic jdbc.clj 1090]
  [clojure.java.jdbc$execute_query_with_params invoke jdbc.clj 1084]
  [clojure.java.jdbc$db_query_with_resultset_STAR_ invokeStatic jdbc.clj 1106]
  [clojure.java.jdbc$db_query_with_resultset_STAR_ invoke jdbc.clj 1093]
  [clojure.java.jdbc$query invokeStatic jdbc.clj 1182]
  [clojure.java.jdbc$query invoke jdbc.clj 1144]
  [clojure.java.jdbc$query invokeStatic jdbc.clj 1160]
  [clojure.java.jdbc$query invoke jdbc.clj 1144]
  [metabase.driver.sql.ddl$jdbc_query invokeStatic ddl.clj 23]
  [metabase.driver.sql.ddl$jdbc_query invoke ddl.clj 20]
  [metabase.driver.mysql.ddl$fn__82880$read_table__82890 invoke ddl.clj 108]
  [metabase.driver.mysql.ddl$fn__82880$fn__82954 invoke ddl.clj 121]
  [metabase.driver.mysql.ddl$fn__82880 invokeStatic ddl.clj 121]
  [metabase.driver.mysql.ddl$fn__82880 invoke ddl.clj 78]
  [clojure.lang.MultiFn invoke MultiFn.java 229]
  [metabase.api.database$fn__76531 invokeStatic database.clj 691]
  [metabase.api.database$fn__76531 invoke database.clj 679]
  [compojure.core$wrap_response$fn__28102 invoke core.clj 160]
  [compojure.core$wrap_route_middleware$fn__28086 invoke core.clj 132]
  [compojure.core$wrap_route_info$fn__28091 invoke core.clj 139]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 151]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 152]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 152]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 152]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 152]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 152]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 152]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 152]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114 invoke core.clj 200]
  [metabase.server.middleware.auth$enforce_authentication$fn__68945 invoke auth.clj 14]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114 invoke core.clj 200]
  [compojure.core$make_context$handler__28142 invoke core.clj 289]
  [compojure.core$make_context$fn__28146 invoke core.clj 299]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$make_context$fn__28146 invoke core.clj 300]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$make_context$fn__28146 invoke core.clj 300]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$make_context$fn__28146 invoke core.clj 300]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$make_context$fn__28146 invoke core.clj 300]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$make_context$fn__28146 invoke core.clj 300]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$make_context$fn__28146 invoke core.clj 300]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$make_context$fn__28146 invoke core.clj 300]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [metabase.api.routes$fn__79930$fn__79933 invoke routes.clj 60]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114 invoke core.clj 200]
  [clojure.lang.AFn applyToHelper AFn.java 160]
  [clojure.lang.AFn applyTo AFn.java 144]
  [clojure.core$apply invokeStatic core.clj 667]
  [clojure.core$apply invoke core.clj 662]
  [metabase.server.routes$fn__80085$fn__80086 doInvoke routes.clj 67]
  [clojure.lang.RestFn invoke RestFn.java 436]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114 invoke core.clj 200]
  [compojure.core$make_context$handler__28142 invoke core.clj 289]
  [compojure.core$make_context$fn__28146 invoke core.clj 299]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke core.clj 153]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke core.clj 197]
  [metabase.server.routes$fn__80071$fn__80073 invoke routes.clj 45]
  [compojure.core$routes$fn__28114$f__28115 invoke core.clj 198]
  [compojure.core$routes$fn__28114 invoke core.clj 200]
  [metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__76822 invoke exceptions.clj 98]
  [metabase.server.middleware.exceptions$catch_api_exceptions$fn__76819 invoke exceptions.clj 86]
  [metabase.server.middleware.log$log_api_call$fn__80536$fn__80537$fn__80538 invoke log.clj 211]
  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invokeStatic diagnostic.clj 15]
  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invoke diagnostic.clj 9]
  [metabase.server.middleware.log$log_api_call$fn__80536$fn__80537 invoke log.clj 203]
  [toucan.db$_do_with_call_counting invokeStatic db.clj 236]
  [toucan.db$_do_with_call_counting invoke db.clj 229]
  [metabase.server.middleware.log$log_api_call$fn__80536 invoke log.clj 202]
  [metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__83637 invoke browser_cookie.clj 38]
  [metabase.server.middleware.security$add_security_headers$fn__62337 invoke security.clj 149]
  [metabase.server.middleware.json$wrap_json_body$fn__82594 invoke json.clj 63]
  [metabase.server.middleware.json$wrap_streamed_json_response$fn__82612 invoke json.clj 99]
  [metabase.server.middleware.offset_paging$handle_paging$fn__62361 invoke offset_paging.clj 42]
  [ring.middleware.keyword_params$wrap_keyword_params$fn__83904 invoke keyword_params.clj 55]
  [ring.middleware.params$wrap_params$fn__83923 invoke params.clj 77]
  [metabase.server.middleware.misc$maybe_set_site_url$fn__35776 invoke misc.clj 59]
  [metabase.server.middleware.session$reset_session_timeout$fn__47465 invoke session.clj 377]
  [metabase.server.middleware.session$bind_current_user$fn__47448$fn__47449 invoke session.clj 313]
  [metabase.server.middleware.session$do_with_current_user invokeStatic session.clj 292]
  [metabase.server.middleware.session$do_with_current_user invoke session.clj 281]
  [metabase.server.middleware.session$bind_current_user$fn__47448 invoke session.clj 312]
  [metabase.server.middleware.session$wrap_current_user_info$fn__47430 invoke session.clj 262]
  [metabase.server.middleware.session$wrap_session_id$fn__47414 invoke session.clj 195]
  [metabase.server.middleware.auth$wrap_api_key$fn__68953 invoke auth.clj 27]
  [ring.middleware.cookies$wrap_cookies$fn__83824 invoke cookies.clj 216]
  [metabase.server.middleware.misc$add_content_type$fn__35759 invoke misc.clj 27]
  [metabase.server.middleware.misc$disable_streaming_buffering$fn__35784 invoke misc.clj 76]
  [ring.middleware.gzip$wrap_gzip$fn__83866 invoke gzip.clj 86]
  [metabase.server.middleware.misc$bind_request$fn__35787 invoke misc.clj 93]
  [metabase.server.middleware.ssl$redirect_to_https_middleware$fn__83653 invoke ssl.clj 38]
  [metabase.server$async_proxy_handler$fn__80308 invoke server.clj 73]
  [metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a handle nil -1]
  [org.eclipse.jetty.server.handler.HandlerWrapper handle HandlerWrapper.java 127]
  [org.eclipse.jetty.server.Server handle Server.java 516]
  [org.eclipse.jetty.server.HttpChannel lambda$handle$1 HttpChannel.java 487]
  [org.eclipse.jetty.server.HttpChannel dispatch HttpChannel.java 732]
  [org.eclipse.jetty.server.HttpChannel handle HttpChannel.java 479]
  [org.eclipse.jetty.server.HttpConnection onFillable HttpConnection.java 277]
  [org.eclipse.jetty.io.AbstractConnection$ReadCallback succeeded AbstractConnection.java 311]
  [org.eclipse.jetty.io.FillInterest fillable FillInterest.java 105]
  [org.eclipse.jetty.io.ChannelEndPoint$1 run ChannelEndPoint.java 104]
  [org.eclipse.jetty.util.thread.QueuedThreadPool runJob QueuedThreadPool.java 883]
  [org.eclipse.jetty.util.thread.QueuedThreadPool$Runner run QueuedThreadPool.java 1034]
  [java.lang.Thread run Thread.java 829]]}
2022-09-21 18:22:36,042 WARN mysql.ddl :: Undoing step `:persist.check/create-table` for db DELETE
2022-09-21 18:22:36,043 WARN mysql.ddl :: Undoing step `:persist.check/create-schema` for db DELETE
2022-09-21 18:22:36,168 ERROR middleware.log :: POST /api/database/93/persist 500 21.1 ms (3 DB calls) 
{:via
 [{:type clojure.lang.ExceptionInfo,
   :message "Lack permission to read table in schema metabase_cache_f74b7_93",
   :data {:error :persist.check/read-table, :database "DELETE"},
   :at [metabase.api.database$fn__76531 invokeStatic "database.clj" 701]}],
 :trace
 [[metabase.api.database$fn__76531 invokeStatic "database.clj" 701]
  [metabase.api.database$fn__76531 invoke "database.clj" 679]
  [compojure.core$wrap_response$fn__28102 invoke "core.clj" 160]
  [compojure.core$wrap_route_middleware$fn__28086 invoke "core.clj" 132]
  [compojure.core$wrap_route_info$fn__28091 invoke "core.clj" 139]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 151]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 152]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 152]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 152]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 152]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 152]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 152]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 152]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114 invoke "core.clj" 200]
  [metabase.server.middleware.auth$enforce_authentication$fn__68945 invoke "auth.clj" 14]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114 invoke "core.clj" 200]
  [compojure.core$make_context$handler__28142 invoke "core.clj" 289]
  [compojure.core$make_context$fn__28146 invoke "core.clj" 299]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$make_context$fn__28146 invoke "core.clj" 300]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$make_context$fn__28146 invoke "core.clj" 300]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$make_context$fn__28146 invoke "core.clj" 300]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$make_context$fn__28146 invoke "core.clj" 300]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$make_context$fn__28146 invoke "core.clj" 300]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$make_context$fn__28146 invoke "core.clj" 300]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$make_context$fn__28146 invoke "core.clj" 300]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [metabase.api.routes$fn__79930$fn__79933 invoke "routes.clj" 60]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114 invoke "core.clj" 200]
  [clojure.lang.AFn applyToHelper "AFn.java" 160]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 667]
  [clojure.core$apply invoke "core.clj" 662]
  [metabase.server.routes$fn__80085$fn__80086 doInvoke "routes.clj" 67]
  [clojure.lang.RestFn invoke "RestFn.java" 436]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114 invoke "core.clj" 200]
  [compojure.core$make_context$handler__28142 invoke "core.clj" 289]
  [compojure.core$make_context$fn__28146 invoke "core.clj" 299]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [compojure.core$wrap_route_matches$fn__28095 invoke "core.clj" 153]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114$f__28115$respond_SINGLEQUOTE___28116 invoke "core.clj" 197]
  [metabase.server.routes$fn__80071$fn__80073 invoke "routes.clj" 45]
  [compojure.core$routes$fn__28114$f__28115 invoke "core.clj" 198]
  [compojure.core$routes$fn__28114 invoke "core.clj" 200]
  [metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__76822 invoke "exceptions.clj" 98]
  [metabase.server.middleware.exceptions$catch_api_exceptions$fn__76819 invoke "exceptions.clj" 86]
  [metabase.server.middleware.log$log_api_call$fn__80536$fn__80537$fn__80538 invoke "log.clj" 211]
  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invokeStatic "diagnostic.clj" 15]
  [metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info invoke "diagnostic.clj" 9]
  [metabase.server.middleware.log$log_api_call$fn__80536$fn__80537 invoke "log.clj" 203]
  [toucan.db$_do_with_call_counting invokeStatic "db.clj" 236]
  [toucan.db$_do_with_call_counting invoke "db.clj" 229]
  [metabase.server.middleware.log$log_api_call$fn__80536 invoke "log.clj" 202]
  [metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__83637 invoke "browser_cookie.clj" 38]
  [metabase.server.middleware.security$add_security_headers$fn__62337 invoke "security.clj" 149]
  [metabase.server.middleware.json$wrap_json_body$fn__82594 invoke "json.clj" 63]
  [metabase.server.middleware.json$wrap_streamed_json_response$fn__82612 invoke "json.clj" 99]
  [metabase.server.middleware.offset_paging$handle_paging$fn__62361 invoke "offset_paging.clj" 42]
  [ring.middleware.keyword_params$wrap_keyword_params$fn__83904 invoke "keyword_params.clj" 55]
  [ring.middleware.params$wrap_params$fn__83923 invoke "params.clj" 77]
  [metabase.server.middleware.misc$maybe_set_site_url$fn__35776 invoke "misc.clj" 59]
  [metabase.server.middleware.session$reset_session_timeout$fn__47465 invoke "session.clj" 377]
  [metabase.server.middleware.session$bind_current_user$fn__47448$fn__47449 invoke "session.clj" 313]
  [metabase.server.middleware.session$do_with_current_user invokeStatic "session.clj" 292]
  [metabase.server.middleware.session$do_with_current_user invoke "session.clj" 281]
  [metabase.server.middleware.session$bind_current_user$fn__47448 invoke "session.clj" 312]
  [metabase.server.middleware.session$wrap_current_user_info$fn__47430 invoke "session.clj" 262]
  [metabase.server.middleware.session$wrap_session_id$fn__47414 invoke "session.clj" 195]
  [metabase.server.middleware.auth$wrap_api_key$fn__68953 invoke "auth.clj" 27]
  [ring.middleware.cookies$wrap_cookies$fn__83824 invoke "cookies.clj" 216]
  [metabase.server.middleware.misc$add_content_type$fn__35759 invoke "misc.clj" 27]
  [metabase.server.middleware.misc$disable_streaming_buffering$fn__35784 invoke "misc.clj" 76]
  [ring.middleware.gzip$wrap_gzip$fn__83866 invoke "gzip.clj" 86]
  [metabase.server.middleware.misc$bind_request$fn__35787 invoke "misc.clj" 93]
  [metabase.server.middleware.ssl$redirect_to_https_middleware$fn__83653 invoke "ssl.clj" 38]
  [metabase.server$async_proxy_handler$fn__80308 invoke "server.clj" 73]
  [metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a handle nil -1]
  [org.eclipse.jetty.server.handler.HandlerWrapper handle "HandlerWrapper.java" 127]
  [org.eclipse.jetty.server.Server handle "Server.java" 516]
  [org.eclipse.jetty.server.HttpChannel lambda$handle$1 "HttpChannel.java" 487]
  [org.eclipse.jetty.server.HttpChannel dispatch "HttpChannel.java" 732]
  [org.eclipse.jetty.server.HttpChannel handle "HttpChannel.java" 479]
  [org.eclipse.jetty.server.HttpConnection onFillable "HttpConnection.java" 277]
  [org.eclipse.jetty.io.AbstractConnection$ReadCallback succeeded "AbstractConnection.java" 311]
  [org.eclipse.jetty.io.FillInterest fillable "FillInterest.java" 105]
  [org.eclipse.jetty.io.ChannelEndPoint$1 run "ChannelEndPoint.java" 104]
  [org.eclipse.jetty.util.thread.QueuedThreadPool runJob "QueuedThreadPool.java" 883]
  [org.eclipse.jetty.util.thread.QueuedThreadPool$Runner run "QueuedThreadPool.java" 1034]
  [java.lang.Thread run "Thread.java" 829]],
 :cause "Lack permission to read table in schema metabase_cache_f74b7_93",
 :data {:error :persist.check/read-table, :database "DELETE"},
 :message "Lack permission to read table in schema metabase_cache_f74b7_93",
 :error :persist.check/read-table,
 :database "DELETE"}
```

</details>

**Information about your Metabase Installation:**
Tested 0.44.3

**Additional context**
As noted in https://discourse.metabase.com/t/turn-model-caching-on-functionality-not-working/21624/5?u=flamber, it is likely caused by MySQL not allowing creating table with that syntax. On Postgres a different syntax is used, so perhaps we should try to find the most generic backward compatible way, so it can be used for all/most database types without them creating a specific driver specific functions.

--- a/src/metabase/driver/mysql/ddl.clj
+++ b/src/metabase/driver/mysql/ddl.clj
@@ -87,7 +87,7 @@
                                               (map :schema_name)
                                               (into #{}))]
                     (or (contains? existing-schemas schema-name)
-                      (sql.ddl/execute! conn [(sql.ddl/create-schema-sql database)]))))
+                        (sql.ddl/execute! conn [(sql.ddl/create-schema-sql database)]))))
                 (fn undo-check-schema [conn]
                   (sql.ddl/execute! conn [(sql.ddl/drop-schema-sql database)]))]
                [:persist.check/create-table
@@ -96,11 +96,11 @@
                                          db-spec
                                          (.toMillis (t/minutes 10))
                                          [(sql.ddl/create-table-sql
-                                            database
-                                            {:table-name table-name
-                                             :field-definitions [{:field-name "field"
-                                                                  :base-type :type/Text}]}
-                                            "values (1)")]))
+                                           database
+                                           {:table-name table-name
+                                            :field-definitions [{:field-name "field"
+                                                                 :base-type :type/Text}]}
+                                           "values (1)")]))
                 (fn undo-create-table [conn]
                   (sql.ddl/execute! conn [(sql.ddl/drop-table-sql database table-name)]))]
                [:persist.check/read-table
@@ -116,27 +116,27 @@
     ;; Unlike postgres, mysql ddl clauses will not rollback in a transaction.
     ;; So we keep track of undo-steps to manually rollback previous, completed steps.
     (jdbc/with-db-connection [conn db-spec]
-                             (loop [[[step stepfn undofn] & remaining] steps
-                                    undo-steps []]
-                               (let [result (try (stepfn conn)
-                                                 (log/info (trs "Step {0} was successful for db {1}"
-                                                                step (:name database)))
-                                                 ::valid
-                                                 (catch Exception e
-                                                   (log/warn (trs "Error in `{0}` while checking for model persistence permissions." step))
-                                                   (log/warn e)
-                                                   (try
-                                                     (doseq [[undo-step undofn] (reverse undo-steps)]
-                                                       (log/warn (trs "Undoing step `{0}` for db {1}" undo-step (:name database)))
-                                                       (undofn conn))
-                                                     (catch Exception _e
-                                                       (log/warn (trs "Unable to rollback database check for model persistence"))))
-                                                   step))]
-                                 (cond (and (= result ::valid) remaining)
-                                       (recur remaining (conj undo-steps [step undofn]))
+      (loop [[[step stepfn undofn] & remaining] steps
+             undo-steps []]
+        (let [result (try (stepfn conn)
+                          (log/info (trs "Step {0} was successful for db {1}"
+                                         step (:name database)))
+                          ::valid
+                          (catch Exception e
+                            (log/warn (trs "Error in `{0}` while checking for model persistence permissions." step))
+                            (log/warn e)
+                            (try
+                              (doseq [[undo-step undofn] (reverse undo-steps)]
+                                (log/warn (trs "Undoing step `{0}` for db {1}" undo-step (:name database)))
+                                (undofn conn))
+                              (catch Exception _e
+                                (log/warn (trs "Unable to rollback database check for model persistence"))))
+                            step))]
+          (cond (and (= result ::valid) remaining)
+                (recur remaining (conj undo-steps [step undofn]))
 
-                                       (= result ::valid)
-                                       [true :persist.check/valid]
+                (= result ::valid)
+                [true :persist.check/valid]
 
-                                       :else
-                                       [false step]))))))
+                :else
+                [false step]))))))

--- a/src/metabase/driver/mysql/ddl.clj
+++ b/src/metabase/driver/mysql/ddl.clj
@@ -100,7 +100,7 @@
                                            {:table-name table-name
                                             :field-definitions [{:field-name "field"
                                                                  :base-type :type/Text}]}
-                                           "values (1)")]))
+                                           "select 1")]))
                 (fn undo-create-table [conn]
                   (sql.ddl/execute! conn [(sql.ddl/drop-table-sql database table-name)]))]
                [:persist.check/read-table

--- a/test/metabase/analytics/prometheus_test.clj
+++ b/test/metabase/analytics/prometheus_test.clj
@@ -132,7 +132,7 @@
                                          nil)
             _              (assert c3p0-collector "Did not find c3p0 collector")
             measurements   (.collect ^Collector c3p0-collector)
-            _              (is (pos? (doto (count measurements) tap>))
+            _              (is (pos? (count measurements))
                                "No measurements taken")]
         (is (= (count (:connection-pools (troubleshooting/connection-pool-info)))
                (count (.samples ^GaugeMetricFamily (first measurements))))

--- a/test/metabase/task/persist_refresh_test.clj
+++ b/test/metabase/task/persist_refresh_test.clj
@@ -3,6 +3,8 @@
             [clojurewerkz.quartzite.conversion :as qc]
             [java-time :as t]
             [medley.core :as m]
+            [metabase.driver :as driver]
+            [metabase.driver.ddl.interface :as ddl.i]
             [metabase.models :refer [Card Database PersistedInfo TaskHistory]]
             [metabase.query-processor :as qp]
             [metabase.query-processor.interface :as qp.i]
@@ -218,3 +220,12 @@
                                     :query    {:aggregation [[:count]]
                                                :source-table (str "card__" (:id model))}}]
                   (is (= [[num-rows-query]] (mt/rows (qp/process-query query-on-top)))))))))))))
+
+(deftest can-persist-test
+  (testing "Can each database that allows for persistence actually persist"
+    (mt/test-drivers (mt/normal-drivers-with-feature :persist-models)
+      (testing (str driver/*driver* " can persist")
+        (mt/dataset test-data
+          (let [[success? error] (ddl.i/check-can-persist (mt/db))]
+            (is success? (str "Not able to persist on " driver/*driver*))
+            (is (= :persist.check/valid error))))))))

